tcmu-runner.service

```diff
@@ -1,5 +1,6 @@
 [Unit]
 Description=LIO Userspace-passthrough daemon
+After=glusterd.service
```

Comment:
Does this affect deployments that don't have a glusterd.service? Perhaps the glusterd.service should just have a ```Before=tcmu-runner```?

Reply:
@pkalever I saw you comment on the other systemd PR. Just wanted to make sure you saw Jason's comment. I am just waiting for a reply.

We added Wants/Afters in the ceph services that use tcmu-runner and it works for us. Our design might be different though, so let us know.

Reply:
@mikechristie @dillaman sorry, looks like I have missed this one.

@dillaman nope, IMO this will not affect deployments which doesn't have gluster. Because "After=" will not start any unit. This just helps in sorting the dependencies chain if a parent unit starts both tcmu-runner and glusterd. 

For me it is more meaning full to add storage units as dependencies to tcmu-runner in the tcmu-runner unit because ceph and gluster are reasonably a layer below tcmu-runner. Because tcmu-runner needs storage before itself gets up, it is important that we defend on the precedence in the runner unit. More context at https://github.com/open-iscsi/tcmu-runner/pull/301

Reply:
@mikechristie @dillaman 

I have test the "After=glusterd.service" at #301 case, and without gluster.service locally, it works fine as @pkalever mentioned above.

Reply:
@lxbsz Thanks!

Reply:
@pkalever what do you guys use for the target initialization? Do you use the target.service to start lio? To setup targets and map luns to gluster block devs, do you rely on target.service and it reading the /etc/target info or is that done by something in the glusterd.service?

Reply:
@mikechristie We rely on target.servic.

Here is the sequence that we follow.
It is a Prerequisite that glusterd.service should be running.

On a start of gluster-blockd.service (distributed block management framework) gluster-blockd will trigger gluster-block-target.service (wrapper over target.service) which intern trigger tcmu-runner and start itself after tcmu-runner comes up. Once tcmu-runner gluster-block-target comes up gluster-blockd will get started.

We have Requisite=glusterd.service (intentional) in gluster-block-target.service and gluster-blockd.service, which means just check if glusterd is up, in case it is up we go further else we will fail the entire transaction. Unlike Requires, Requisite will not trigger the listed service (glusterd).

Reply:
Just to make sure I am on the same page you do not actually need this change, because the gluster unit files have the necessary After/Before settings. You just want this patch to be defensive.

https://github.com/gluster/gluster-block/blob/master/systemd/gluster-blockd.service.in

has

After=gluster-block-target.service

so it will start after

https://github.com/gluster/gluster-block/blob/master/systemd/gluster-block-target.service.in

which has

After=glusterd.service tcmu-runner.service

so when gluster-block-target and gluster-blockd are up, glusterd and tcmu-runner are going to be up.